Change post-invite success message in case when user invites only one…

TomaszKolek opened the pull request:
… user.

Fixes: #1412.

timabbott commented:
Can you add tests for the email parsing code?  

Also, the pluralization code isn't internationalized.  See http://zulip.readthedocs.io/en/latest/translating.html#frontend-translations for notes on how to do the plurals properly; @umairwaheed can help if anything in those docs isn't clear (they're quite new!).

umairwaheed commented:
Sure!

TomaszKolek commented:
@umairwaheed 
Please take a look at my changes - I'm not sure if its done right.
Where can I find resource file to add `user` word?

timabbott commented:
It looks like code in the two commits are a bit mixed up?

TomaszKolek commented:
@timabbott I don't know what was wrong - Fixed.

umairwaheed commented on static/js/invite.js:
```diff
@@ -65,9 +65,10 @@ exports.initialize = function () {
-            invite_status.text('User' + additional_s_if_needed + ' invited successfully.')
+            var user_word = i18n.t('user', {count: (invitee_emails.val().match(/@/g) || [] ).length});
+            user_word = user_word[0].toUpperCase() + user_word.slice(1);
+            invite_status.text(user_word + ' invited successfully.')
```
@TomaszKolek, I think we should mark the whole sentence for translation:
```
invite_status.text(i18n.t('User invited successfully.', {'count': (invitee_emails.val().match(/@/g) || [] ).length}));
```

Also, please add the following key-value pairs in `locale/static/en/translations.json`:
```
{
...
"User invited successfully.": "User invited successfully.",
"User invited successfully._plural": "Users invited successfully."
...
}
```

This way you won't have to add any business logic; i18next will handle the plurals on its own.

TomaszKolek commented:
@umairwaheed
So I did as you wrote but probably did something wrong.
I added records to translations.json and actually copied your code.
Unfortunately, it does not work.

I also added as a test record with `a` key and normal content. 
When I did something like:
`invite_status.text(i18n.t('a', {'count': (invitee_emails.val().match(/@/g) || [] ).length}));`

I saw just `a` in the box.
Probably I did something wrong there. Could you help me once again please ;-)

timabbott commented:
@TomaszKolek it looks like you squashed the two commits together -- can you split out the email parsing one again?  Thanks!

TomaszKolek commented:
@timabbott I don't know what I did with it... Should be ok now.

timabbott commented:
For the tests, I think in addition to the ones you have, we should move the `invitee_emails_list` extraction into a function, and write unit tests for that function with various input strings.  I think that will do a much better job of confirming that this logic works correctly in all cases.

(Also, from reading the code, I think the current logic `re.split` expression isn't correct for email addresses in that format -- see this test:
```
re.split(r'[, \n]', 'Tim Abbott <tabbott@mit.edu>')
['Tim', 'Abbott', '<tabbott@mit.edu>']
```
)

timabbott commented:
(And actually that's why the current tests don't pass)

umairwaheed commented:
@TomaszKolek, your code is working on my side :). Can you please debug the expression that gives value to `count`?
![screen shot 2016-07-30 at 8 35 05 pm](https://cloud.githubusercontent.com/assets/641763/17271442/2c69dd22-5695-11e6-8510-2cf6491688ba.png)

TomaszKolek commented:
@umairwaheed 
I don't know what I did wrong - it works now.

Thank you!

umairwaheed commented:
Awesome! No problem.

TomaszKolek commented:
@timabbott Tests added.

timabbott commented:
Awesome, thanks @TomaszKolek!  I merged these, with a change to the commit message of one of the commits to: 64b5d0ce647d290c4177896e1b61c8923c0be17e.

The main commit message changes were:
* Adding a blank line after the first line of the commit message
* Providing more explanation for the problem being solved / previous behavior.

Thanks @TomaszKolek!
